packages/menu/src/MenuItem.ts

```diff
@@ -276,6 +324,108 @@ export class MenuItem extends LikeAnchor(Focusable) {
         }
     }
 
+    public closeOverlay?: (leave?: boolean) => Promise<void>;
+
+    protected handleSubmenuClick(): void {
+        this.openOverlay({ immediate: true });
+    }
+
+    protected handlePointerenter(): void {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        } else {
+            this.openOverlay();
+        }
+    }
+
+    protected leaveTimeout?: ReturnType<typeof setTimeout>;
+
+    protected handlePointerleave(): void {
+        if (this.hasSubmenu && this.open) {
+            this.active = true;
+            this.leaveTimeout = setTimeout(() => {
+                delete this.leaveTimeout;
+                if (this.closeOverlay) this.closeOverlay(true);
+            }, POINTERLEAVE_TIMEOUT);
+        }
+    }
+
+    protected handleSubmenuChange = (): void => {
+        this.click();
+    };
+
+    protected handleSubmenuPointerenter = (): void => {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        }
+    };
+
+    public async openOverlay({
+        immediate,
+    }: { immediate?: boolean } = {}): Promise<void> {
+        if (this.hasSubmenu && !this.open) {
+            this.open = true;
+            this.submenu = (
+                this.shadowRoot.querySelector(
+                    'slot[name="submenu"]'
+                ) as HTMLSlotElement
+            ).assignedElements()[0] as Menu;
+            this.submenu.addEventListener(
+                'pointerenter',
+                this.handleSubmenuPointerenter
+            );
+            this.submenu.addEventListener('change', this.handleSubmenuChange);
+            const popover = document.createElement('sp-popover');
+            popover.append(this.submenu);
+            this.submenu.tabIndex = 0;
+            this.submenu.removeAttribute('slot');
+            this.menuData.focusRoot?.submenuOpened(this);
+            const closeOverlay = openOverlay(this, 'click', popover, {
+                placement: this.isLTR ? 'right-start' : 'left-start',
+                receivesFocus: 'auto',
+                delayed: !immediate && false,
```

Comment:
I'm not sure I understand what `!immediate && false` is doing here - will this not always evaluate to `delayed: false`?

Reply:
I think I _wanted_ to use delayed, but then realized it's kind of a PITA here and turned it off, but didn't remove it. Removed it for now.